File: bicycle/transportation.py
class Transportation:
    def __init__(self, started : bool = False):
        self._started : bool = started
    @property
    def started(self) -> bool:
        return self._started

    def start(self) -> None:
        self._started = True

    def stop(self) -> None:
        self._started = False

    @property
    def going(self) -> bool:
        return self._started

File: cpp-diamond/src/Transportation.cpp
#include "Transportation.h"

namespace go {

    Transportation::Transportation()
      : started(false)
    { 
    }
    void Transportation::start() {
        started = true;
    }
    void Transportation::stop() {
        started = false;
    }

    bool Transportation::isStarted() const {
        return started;
    }
    bool Transportation::isGoing() const {
        return started;
    }

    Transportation::~Transportation() {
    }

 }

